docs/markdown/snippets/structured_sources.md

```diff
@@ -0,0 +1,25 @@
+## structured_sources()
+
+A new function, `structured_sources()` has been added. This function allows
+languages like Rust which depend on the filesystem layout at compile time to mix
+generated and static sources.
+
+```meson
+executable(
+  'main',
+  structured_sources(
+    'main.rs,
+    mod : generated_mod_rs,
```

Comment:
Might be useful to put `mod` in quotes.

Reply:
Oh, even better. That's still got the old *non working* syntax :(